### PR TITLE
Laboratory work #1, Shuvar Alena - 19FPL2

### PR DESCRIPTION
**Pull request by alenashuvar**

### CODE REVIEW

**Comment by WhiteJaeger** [comment]
The deadline is not that far - consider pushing changes so that we can review them.

File: lab_1/main.py [hunk 1]
```diff
@@ -12,7 +13,13 @@ def tokenize(text: str) -> list:
     e.g. text = 'The weather is sunny, the man is happy.'
     --> ['the', 'weather', 'is', 'sunny', 'the', 'man', 'is', 'happy']
     """
-    pass
+    if type(text) != str:
```
Comment (WhiteJaeger) [inline lab_1/main.py]: `type` is usually considered a bad style, look for smth else.

File: lab_1/main.py [hunk 2]
```diff
@@ -12,7 +13,13 @@ def tokenize(text: str) -> list:
...
+        return []
+    text = text.lower()
```
Comment (WhiteJaeger) [inline lab_1/main.py]: Usually it is better to create a separate variable to work with it and leave input as it is.

File: lab_1/main.py [hunk 3]
```diff
@@ -25,7 +32,16 @@ def remove_stop_words(tokens: list, stop_words: list) -> list:
     stop_words = ['the', 'is']
     --> ['weather', 'sunny', 'man', 'happy']
     """
-    pass
+    if type(tokens) != list:
+        return []
+    elif type(stop_words) != list:
```
Comment (WhiteJaeger) [inline lab_1/main.py]: Please, try to shorten the checks.

File: lab_1/main.py [hunk 4]
```diff
@@ -49,7 +77,19 @@ def get_top_n_words(freq_dict: dict, top_n: int) -> list:
...
-    pass
+    if (type(freq_dict) != dict) or (type(top_n) != int):
+        return []
+    list_dictionary_items = list(freq_dict.items())
+    list_dictionary_items.sort(key=lambda k_v: k_v[1], reverse=True)
+    top_n_words = []
+    if top_n > len(list_dictionary_items):
+        for index in range(len(list_dictionary_items)):
+            top_n_words.append(list_dictionary_items[index][0])
+    else:
```
Comment (WhiteJaeger) [inline lab_1/main.py]: You can get rid of this `else`-block.

File: lab_1/main.py [hunk 5]
```diff
@@ -69,7 +109,29 @@ def get_concordance(tokens: list, word: str, left_context_size: int, right_conte
...
+        return []
+    elif word not in tokens:
+        return []
+    elif (type(left_context_size) != int) and (type(right_context_size) != int):
+        return []
+    elif (left_context_size < 1) and (right_context_size < 1):
+        return []
+    for index, token in enumerate(tokens):
+        if token == word:
+            if (type(left_context_size) != int) or (left_context_size < 1):
```
Comment (WhiteJaeger) [inline lab_1/main.py]: Checks better to be placed somewhere in the beginning of the function.

File: lab_1/main.py [hunk 6]
```diff
@@ -87,16 +149,44 @@ def get_adjacent_words(tokens: list, word: str, left_n: int, right_n: int) -> li
...
 
 
 def read_from_file(path_to_file: str) -> str:
     """
     Opens the file and reads its content
     :return: the initial text in string format
     """
-    with open(path_to_file, 'r', encoding='utf-8') as fs:
-        data = fs.read()
+    with open(path_to_file, 'r', encoding='utf-8') as file:
```
Comment (WhiteJaeger) [inline lab_1/main.py]: It is usually considered a bad style (and a dangerous one!) to use built-in words (e.g. `str`, `list` etc.) as variable names.

File: lab_1/main.py [hunk 7]
```diff
@@ -105,7 +195,14 @@ def write_to_file(path_to_file: str, content: list):
     """
     Writes the result in a file
     """
-    pass
+    context_list = []
+    for word_list in content:
+        s = ' '.join(word_list)
+        context_list.append(s)
+    context_str = '\n'.join(context_list)
+    name_file = path_to_file + 'report.txt'
```
Comment (WhiteJaeger) [inline lab_1/main.py]: Path to file is passed as the following:
`path/to/file.txt` -> no need for this line.

**Review by WhiteJaeger — CHANGES_REQUESTED** [review]

File: lab_1/main.py (hunk 3 above)
Comment (alenashuvar) [reply lab_1/main.py]: I don't know how :(

File: lab_1/main.py [hunk 8]
```diff
@@ -12,7 +13,14 @@ def tokenize(text: str) -> list:
     e.g. text = 'The weather is sunny, the man is happy.'
     --> ['the', 'weather', 'is', 'sunny', 'the', 'man', 'is', 'happy']
     """
-    pass
+    if not isinstance(text, str):
```
Comment (WhiteJaeger) [inline lab_1/main.py]: Nice.

**Comment by WhiteJaeger** [comment]
Great work! You are done here.

**Review by WhiteJaeger — APPROVED** [review]
Great work! You are done here.

**Comment by WhiteJaeger** [comment]
Open another PR with lab_2.

**Pull request closed without merging**